x/gov/handler.go

```diff
@@ -304,25 +339,122 @@ func (h TextProposalHandler) validate(ctx weave.Context, db weave.KVStore, tx we
 	return &msg, rules, elect, nil
 }
 
+type ElectorateUpdateProposalHandler struct {
+	auth        x.Authenticator
+	elecBucket  *ElectorateBucket
+	propBucket  *ProposalBucket
+	rulesBucket *ElectionRulesBucket
+}
+
+func (h ElectorateUpdateProposalHandler) Check(ctx weave.Context, db weave.KVStore, tx weave.Tx) (*weave.CheckResult, error) {
+	if _, _, _, err := h.validate(ctx, db, tx); err != nil {
+		return nil, err
+	}
+	return &weave.CheckResult{GasAllocated: proposalCost}, nil
+
+}
+
+func (h ElectorateUpdateProposalHandler) Deliver(ctx weave.Context, db weave.KVStore, tx weave.Tx) (*weave.DeliverResult, error) {
+	msg, rules, electorate, err := h.validate(ctx, db, tx)
+	if err != nil {
+		return nil, err
+	}
+	blockTime, _ := weave.BlockTime(ctx)
```

Comment:
I wonder if we should check the notation of `weave.BlockTime` to panic if the time is not set. Second argument seems to be ignored by everyone https://github.com/iov-one/weave/search?q=BlockTime&unscoped_q=BlockTime

Reply:
Yes, we had this discussion once. Except for tests it will either be set by genesis time (block 0) or by the consensus engine based on votes timestamps. So it won't be empty. Can you open a new issue?

Reply:
https://github.com/iov-one/weave/issues/582